### ISSUE TITLE
max_seq_length doesn't seem to considered

### ISSUE THREAD

**Issue by weidotwisc** [foundation-model-stack/fms-hf-tuning#232]
## Describe the bug

When I specify ```max_seq_length``` in my running script, this parameter doesn't seem to considered by SFT.

## Platform

CCC

- Interpreter version:
- Python 3.11.0
- Library version:
- I did a ```pip install fms-hf-tuning```, so I think it looks into https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/pyproject.toml and installed required libraries 
- Also I have checked out fms-hf-funning w/ version v0.4.0, hoping to get a stable version.

## Sample Code

(1) Please see attached ```run_weiz.sh```(https://github.com/user-attachments/files/16112739/run_weiz.txt, renamed to .txt because it seems that I cannot upload a .sh file)

(2) In the above-mentioned script, you would just need to replace two variables ```TRAIN_DATA_PATH``` and ```OUTPUT_PATH``` to your file system proper paths. Furthermore, I have uploaded training data (which ```TRAIN_DATA_PATH``` should point to, see below) 

(3) Training data: [citv3.3_200k_python_ccg_sft_tiny.json](https://github.com/user-attachments/files/16112797/citv3.3_200k_python_ccg_sft_tiny.json)

## Expected behavior

```max_seq_length``` should be correctly accounted for.

## Observed behavior
For all the 100 training samples, SFT complains 
```This instance will be ignored in loss calculation. Note, if this happens often, consider increasing the `max_seq_length`.```

## Additional context
@Ssukriti @anhuong For the reference. Thanks!

**Comment by anhuong** [comment]
this was further discussed over Slack and resolved:
> none of these are bugs. the errors are because response template is not found. it is the token values that need to match not string match .
> the token values of your response template are `[78, 96, 10963, 3491, 96]
> the way to debug would be to tokenize the input string and see if these token values are in the string or not
>
> if not, try `\nAnswer:\n   or `\nAnswer: as response template instead. sometimes the token value for \n\nAnswer need not be same as \n---\n\nAnswer that is in the input text
> you can do tokenize(response template) and tokenize(input data) to see if response temnplate is subset of input data or not
>
> it is not very straughtforward to find a response template that will match - that is why are moving away from it in future.
> the reason we recommend alpaca template is because the template is robust to tokenizer changes
>
> we have since worked to implement -- we just ask the users to only come up input and output . In the input and output , a user can format strings however the way they want (e.g., the input will contain things like \n\nAnswer:\n  in the end). And SFT  only needs to mask the loss for input
>
> we have enough proof that max_seq_length works. the reason you are getting that error is because the response template is not found. truncating could be one reason, thats why SFT TRainer returns that generic error . but its not the only reason
